index.bs

```diff
@@ -4572,6 +4572,55 @@ and similarly, closing or aborting the <code>writable</code> side will implicitl
 
 </div>
 
+<h3 id="example-ts-promise-to-readable">Using an identity transform stream as a primitive when manipulating streams</h3>
+
+Combining an identity transform stream with {{pipeTo()}} is a powerful way to manipulate streams. For example,
+concatenating multiple readable streams into one:
+
+<pre><code class="lang-javascript">
+function concatenateReadables(readables) {
```

Comment:
Although we haven't used async functions much so far, in this case they give a pretty clear benefit:

```js
async function concatenateReadables(readables) {
  const ts = new TransformStream();
  
  for (const readable of readables) {
    try {
      await readable.pipeTo(ts.writable, { preventClose: true });
    } catch (e) {
      await Promise.all([
        ts.writable.abort(e),
        readable.cancel(e)
      ]);
      return;
    }
  }
  
  await ts.writable.getWriter().close();
  return ts.readable;
}
```

This departs from yours slightly in some error cases in ways that I think are probably improvements?

I'm also unsure that the catch() is necessary. Perhaps reading the tests will illuminate why you included it.

Reply:
Using "async function" makes it return a promise. We could of course wrap it in a call to promiseToReadable(), but that would be poor for ergonomics.